packages/signer-cli/src/cmdSign.ts

```diff
@@ -13,7 +13,34 @@ const curvePrefixes: {[key in Curves]: [number] } = {
   sr25519: [1]
 };
 
+function isHexSeed (seed: string): boolean {
+  return isHex(seed) && seed.length === 66;
+}
+
+function rawValidate (seed: string): boolean {
+  return ((seed.length > 0) && (seed.length <= 32)) || isHexSeed(seed);
+}
+
+/**
+   * seed here can be any of the following:
+   *  1. full SURI: <mnemonic>//<hard>/<soft>///<password>
+   *  2. raw hex with or without derivation path: <hex>//<hard>/<soft>///<password>
+   *  3. just mnemonic: hard clown circus world laugh ...
+  */
+function validate (seed: string): boolean {
+  try {
+    // this catches cases 1, 2
+    const { phrase } = keyExtractSuri(seed);
+
+    return rawValidate(phrase);
```

Comment:
phrase would be of the form (copying an example) `siren coral design acoustic bargain boy print snack follow sock describe ghost` - so has a number of words and certainly allowed to be > 32 bytes. (the 32-byte limit is only fir the first-gen seeds, which is mostly not in use anymore)

Here is an imperfect example that only caters for mnemonics and hex seeds (the old-style seed is not worth supporting, unless you have to... like the apps UI)

https://github.com/polkadot-js/extension/blob/master/packages/extension-base/src/background/handlers/Extension.ts#L248-L263

Reply:
ok so IIUC, ditch the `rawValidate` (catering first-gen seeds), instead bip39 validate the phrase after extracting it from a full suri.

which would check the phrase length, checksum, & entropy fwict. And whether the words are in the wordlist.

I'm not sure if I'm missing something. 

Is validation of the derivation paths necessary? I'm not sure what more other than checking the regex of the overall string would be necessary here.

Also [this check](https://github.com/polkadot-js/extension/blob/2219ab785fbf82ee031a4744f11724a24bc3c643/packages/extension-base/src/background/handlers/Extension.ts#L255) asserts seed is 256 bits but the [doc](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#from-mnemonic-to-seed) says the generated seed should be 512 bits?

Thank you for your patience :)

Reply:
That assert checks the number of words only. The 256/512 comes in when that is converted to a minisecret, so it is the conversion from mnemonic here https://github.com/polkadot-js/common/blob/master/packages/util-crypto/src/mnemonic/toMiniSecret.ts#L23 (Nothing to validate) 

Can leave out the derivation path validation, I believe the extract does do some basic regex checks on it anyway.

Reply:
> Here is an imperfect example that only caters for mnemonics and hex seeds

So since there's no need to support the old style seeds in the signer-cli, I'll copy pasta that example (unless there's another form of seed other than mnemonics and hex?...)

> That assert checks the number of words only.

sorry, I meant the [ line above. ](https://github.com/polkadot-js/extension/blob/2219ab785fbf82ee031a4744f11724a24bc3c643/packages/extension-base/src/background/handlers/Extension.ts#L252)